src/core.ts

```diff
@@ -26,7 +26,7 @@ export class Core {
                 try {
                     response = JSON.parse(response);
                 } catch (e) {
-                    Core.throwError(callBack,'client_error', 500, 'invalid_json', 'invalid json in response. Probably not a ChargeBee response', e);
+                    return Core.throwError(callBack,'client_error', 500, 'invalid_json', 'invalid json in response. Probably not a ChargeBee response', e);
```

Comment:
I think the return should be after throwError as the throwError would return the callback result.

Reply:
In my opinion it is the same, could you please explain why it would be better to return after the callback has been invoked?

Reply:
I don't believe this will solve the scenario where the `response` returned is a JSON string.

For example:

```javascript
const responseAsString = "testing";
const response = JSON.stringify(responseAsString);
//...
response = JSON.parse(response);
// response is now a string
response.http_status_code = status; // this will throw

Reply:
This make a sense, I don't know if is better throwing a new exception or reforming payload as object tout-court.
In any way this patch interrupts the execution of function event after error callback invocation (2021-12-15 scenario).

Reply:
@fgiova-isendu kind of yes - it just makes the return type dependent on the return value of the callback. I usually only prefer to keep return values of one type in a function. Probably better to keep the return as you have it and remove the `return` in the throwError function!

@Rohland & @fgiova I would see the most reasonable to check if response is at that state of type object. and if not also return. As this would indicate a json response in an unknown structure.

```ts
if (typeof response !== 'object') {
 return Core.throwError(
   callBack,
   'client_error', 
   500, 
   'unsupported_json', 
   'unsupported json in response. Probably not a ChargeBee response', 
   e
 );
}
```

Reply:
@wodka 

1. Since we don't have a declared response pattern, we can't assume that a properly encoded array or string is invalid.
2. The code above miss your constraint proposal:

`
const response = null;
if (typeof response !== 'object') {
 return Core.throwError(
   callBack,
   'client_error', 
   500, 
   'unsupported_json', 
   'unsupported json in response. Probably not a ChargeBee response', 
   e
 );
}
[...]
`

Reply:
Just to note that the status code of 500 cannot be assumed if it's not an error? I assumed that is what the `500` above is referring to.

Reply:
so it should then be set to `res.statusCode` instead of 500